DirectProgramming/DPC++FPGA/Tutorials/Features/mem_channels/src/mem_channels.cpp

```diff
@@ -0,0 +1,152 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+#include <chrono>
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities//include/dpc_common.hpp
+#include "dpc_common.hpp"
+
+using namespace sycl;
+
+constexpr size_t vector_size = 1000000; // size of input vectors
+constexpr double kNs = 1e9;             // number of nanoseconds in a second
+
+// Forward declare the kernel name in the global scope.
+// This FPGA best practice reduces name mangling in the optimization reports.
+class VecAdd;
+
+event runVecAdd(sycl::queue &q, const std::vector<int> &a_vec,
+                const std::vector<int> &b_vec, const std::vector<int> &c_vec,
+                std::vector<int> &sum_vec) {
+#if defined(NO_INTERLEAVING)
+  buffer a_buf(a_vec, {property::buffer::mem_channel{1}});
+  buffer b_buf(b_vec, {property::buffer::mem_channel{2}});
+#if defined(FOUR_CHANNELS)
+  buffer c_buf(c_vec, {property::buffer::mem_channel{3}});
+  buffer sum_buf(sum_vec, {property::buffer::mem_channel{4}}); 
+#else
+  buffer c_buf(c_vec, {property::buffer::mem_channel{1}});
+  buffer sum_buf(sum_vec, {property::buffer::mem_channel{2}});
+#endif
+#else
+  buffer a_buf(a_vec);
+  buffer b_buf(b_vec);
+  buffer c_buf(c_vec);
+  buffer sum_buf(sum_vec);
+#endif
+
+  event e = q.submit([&](handler &h) {
+    accessor a(a_buf, h, read_only);
+    accessor b(b_buf, h, read_only);
+    accessor c(c_buf, h, read_only);
+    accessor sum(sum_buf, h, write_only, no_init);
+
+    h.single_task<VecAdd>([=]() [[intel::kernel_args_restrict]] {
+      for (int i = 0; i < vector_size; i++)
+        sum[i] = a[i] + b[i] + c[i];
+    });
+  });
+  return e;
+}
+
+// Initialize the vector from 0 to vector_size - 1
+void initialize_vector(std::vector<int> &a) {
+  for (size_t i = 0; i < a.size(); i++)
+    a.at(i) = i;
```

Comment:
consider using this 
https://en.cppreference.com/w/cpp/algorithm/iota

Reply:
Done.